datafusion/physical-expr/src/intervals/cp_solver.rs

```diff
@@ -1414,4 +1415,130 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_propagate_comparison() {
+        let left = Interval::new(
```

Comment:
It took me some time to work this out, so I figured it might be helpful to leave some comments explaining what is expected. 

```suggestion
        // In the examples below:
        // `left` is unbounded: [?, ?], 
        // `right` is known to be [1000,1000]
        // so `left` < `right` results in no new knowledge of `right` but knowing that `left` is now < 1000:` [?, 1000)
        let left = Interval::new(
```